libs/langchain/langchain/__init__.py

```diff
@@ -44,6 +44,12 @@ def _warn_on_import(name: str, replacement: Optional[str] = None) -> None:
 # Surfaces Deprecation and Pending Deprecation warnings from langchain.
 surface_langchain_deprecation_warnings()
 
+def __setattr__(name, value):
+    print("foo")
+    if name == 'debug' and value is True:
+        warnings.warn("Debug mode is enabled in LangChain", UserWarning)
+    globals()[name] = value
```

Comment:
I also tried this sort of thing and hoped it would work, but no luck.

PEP 562 which defines the module-level `__getattr__()` doesn't allow for a `__setattr__()` so I don't think this ever gets called: https://peps.python.org/pep-0562/

Reply:
ah whoops i tried this out, it didnt work, and i forgot to remove 🤦